.azure/vs2022.yml

```diff
@@ -1302,114 +1325,153 @@ stages:
-               -D CMAKE_C_COMPILER_TARGET=$(platform)-unknown-windows-msvc
-               -D CMAKE_C_FLAGS="/GS- /Oy /Gw /Gy"
-               -D CMAKE_CXX_COMPILER=$(workspace.root)/toolchain-amd64/Library/Developer/Toolchains/unknown-Asserts-development.xctoolchain/usr/bin/clang-cl.exe
-               -D CMAKE_CXX_COMPILER_TARGET=$(platform)-unknown-windows-msvc
-               -D CMAKE_CXX_FLAGS="/GS- /Oy /Gw /Gy"
-               -D CMAKE_MT=mt
-               -D CMAKE_INSTALL_PREFIX=$(Build.StagingDirectory)/Library/Developer/Toolchains/unknown-Asserts-development.xctoolchain/usr
-               -D CMAKE_Swift_COMPILER=$(workspace.root)/toolchain-amd64/Library/Developer/Toolchains/unknown-Asserts-development.xctoolchain/usr/bin/swiftc.exe
-               -D CMAKE_Swift_COMPILER_TARGET=$(platform)-unknown-windows-msvc
-               -D CMAKE_Swift_FLAGS="-resource-dir $(Pipeline.Workspace)/windows-sdk-$(arch)/Library/Developer/Platforms/Windows.platform/Developer/SDKs/Windows.sdk/usr/lib/swift -L$(Pipeline.Workspace)/windows-sdk-$(arch)/Library/Developer/Platforms/Windows.platform/Developer/SDKs/Windows.sdk/usr/lib/swift/windows"
-               -G Ninja
-               -S $(Build.SourcesDirectory)/swift-package-manager
-               -D ArgumentParser_DIR=$(Agent.BuildDirectory)/swift-argument-parser/cmake/modules
-               -D LLBuild_DIR=$(Agent.BuildDirectory)/swift-llbuild/cmake/modules
-               -D SwiftCollections_DIR=$(Agent.BuildDirectory)/swift-collections/cmake/modules
-               -D SwiftCrypto_DIR=$(Agent.BuildDirectory)/swift-crypto/cmake/modules
-               -D SwiftDriver_DIR=$(Agent.BuildDirectory)/swift-driver/cmake/modules
-               -D SwiftSystem_DIR=$(Agent.BuildDirectory)/swift-system/cmake/modules
-               -D TSC_DIR=$(Agent.BuildDirectory)/swift-tools-support-core/cmake/modules
-               -D Yams_DIR=$(Agent.BuildDirectory)/Yams/cmake/modules
-               -D SQLite3_LIBRARY=$(Pipeline.Workspace)/sqlite-$(arch)-3.36.0/Library/sqlite-3.36.0/usr/lib/SQLite3.lib
-               -D SQLite3_INCLUDE_DIR=$(Pipeline.Workspace)/sqlite-$(arch)-3.36.0/Library/sqlite-3.36.0/usr/include
+              cmakeArgs:
+                -B $(Agent.BuildDirectory)/swift-asn1
+                -D BUILD_SHARED_LIBS=NO
+                -D CMAKE_BUILD_TYPE=Release
+                -D CMAKE_MT=mt
+                -D CMAKE_Swift_COMPILER=$(workspace.root)/toolchain-amd64/Library/Developer/Toolchains/unknown-Asserts-development.xctoolchain/usr/bin/swiftc.exe
+                -D CMAKE_Swift_COMPILER_TARGET=$(platform)-unknown-windows-msvc
+                -D CMAKE_Swift_FLAGS="-resource-dir $(Pipeline.Workspace)/windows-sdk-$(arch)/Library/Developer/Platforms/Windows.platform/Developer/SDKs/Windows.sdk/usr/lib/swift -L$(Pipeline.Workspace)/windows-sdk-$(arch)/Library/Developer/Platforms/Windows.platform/Developer/SDKs/Windows.sdk/usr/lib/swift/windows"
```

Comment:
These don't need the `-vfsoverlay`?

Reply:
They do, but the file that I am using for injection is not available across the phases.  So for now we are modifying the VS installation.  We should start using `-sdk` with a staged SDK and avoid the modification of Visual Studio which would then not need the `-vfsoverlay` flag nor modify the installation.  cf. build.ps1